packages/fiori/src/UploadCollectionItem.js

```diff
@@ -0,0 +1,380 @@
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return [ListItem.styles, UploadCollectionItemCss];
+	}
+
+	static get template() {
+		return UploadCollectionItemTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			Button.define(),
+			Icon.define(),
+			Input.define(),
+			Link.define(),
+			Label.define(),
+			fetchI18nBundle("@ui5/webcomponents-fiori"),
+		]);
+	}
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	onBeforeRendering() {
+		if (!this.focused) {
+			this._editing = false;
+		}
+	}
+
+	onAfterRendering() {
+		if (this.focused && this._editing) {
+			this.focusAndSelectText();
+		}
+	}
+
+	async focusAndSelectText() {
+		await this.focus();
+
+		const inp = this.shadowRoot.getElementById("ui5-uci-edit-input");
+
+		if (inp.getFocusDomRef()) {
+			inp.getFocusDomRef().setSelectionRange(0, this._fileNameWithoutExtension.length);
+		}
+	}
+
+	/**
+	 * @override
+	 */
+	onDetailClick(event) {
+		super.onDetailClick(event);
+		this._editing = true;
+	}
+
+	_onInputChange(event) {
+		if (this.shadowRoot.getElementById("ui5-uci-edit-cancel").active) {
+			return;
+		}
+
+		this._editing = false;
+		this.fileName = event.target.value + this._fileExtension;
+		this.fireEvent("rename");
+	}
+
+	_onRenameCancel(event) {
+		this._editing = false;
+	}
+
+	_onFileNameClick(event) {
+		this.fireEvent("fileNameClick");
+	}
+
+	_onRetry(event) {
+		this.fireEvent("retry");
+	}
+
+	_onTerminate(event) {
+		this.fireEvent("terminate");
+	}
+
+	/**
+	 * @override
+	 */
+	get classes() {
+		const result = super.classes;
+
+		return {
+			main: {
+				...result.main,
+				"ui5-uci-root": true,
+				"ui5-uci-root-editing": this._editing,
+				"ui5-uci-root-uploading": this.uploadState === UploadState.Uploading,
+			},
+			progressIndicator: {
+				"ui5-uci-progress-indicator": true,
+				"error": this.uploadState === UploadState.Error,
+			},
+		};
+	}
+
+	get styles() {
+		return {
+			progressBar: {
+				"flex-basis": `${this.progress}%`,
+			},
+			progressBarRemaining: {
+				"border-left": this.progress !== 0 ? "none" : "",
+				"border-radius": this.progress === 0 ? "0.5rem" : "0 0.5rem 0.5rem 0",
+			},
+		};
+	}
+
+	/**
+	 * @override
+	 */
+	get modeDelete() {
+		return !this.noDelete && super.modeDelete;
+	}
+
+	get _fileNameWithoutExtension() {
+		return this.fileName.substring(0, this.fileName.length - this._fileExtension.length);
+	}
+
+	get _fileExtension() {
+		return this.fileName.includes(".") ? `.${this.fileName.split(".").pop()}` : "";
+	}
+
+	get _renameBtnText() {
+		return this.i18nBundle.getText(UPLOADCOLLECTIONITEM_RENAMEBUTTON_TEXT);
+	}
+
+	get _cancelRenameBtnText() {
+		return this.i18nBundle.getText(UPLOADCOLLECTIONITEM_CANCELBUTTON_TEXT);
+	}
+
+	get _showProgressIndicator() {
+		return this.uploadState !== UploadState.Complete;
+	}
+
+	get _progressText() {
+		switch (this.uploadState) {
```

Comment:
switch is an overkill here
if (error) {
  return
}

if (warning) {
 return
}

return... ;

Reply:
ESLint doesn't let me fix the indentation, but you are right that it is too much to use switch-case here. I have turned it to ifs